### PR TITLE
Adopt the Python3-Pip-Skeleton 

### LINKED ISSUE #28: CI and automated builds
The Travis CI is no longer running, and so there is no automated build or testing of cothread. We should create a CI for GitHub.

Other DLS Python modules are moving to use [this skeleton project ](https://github.com/dls-controls/dls-python3-skeleton) as a template. It comes with CI for both code and documentation builds, as well as automating uploads to PyPi. It should provide a solid framework, although tweaks will be required as cothread does have some differences to a pure python module.

### PR DESCRIPTION
**Pull request by AlexanderWells-diamond**
This PR adopts the [python3-pip-skeleton ](https://github.com/DiamondLightSource/python3-pip-skeleton). Note that this PR includes references to this project existing in the [DiamondLightSource Organization](https://github.com/DiamondLightSource). Once this PR is merged I will migrate this repo to that organization.

The (major) changes here include:
- Overhauling the GitHub actions to automatically run tests, publish new tags to PyPi, and build+release docs (note this will use GitHub Pages, not readthedocs)
- Deleting many unused files
- Adding the "tox" tool that can be used to build docs, run tests, and run some pre-commit checks
- Add some default VSCode configuration
- Change the versioning to be based on tags, using setuptools_scm. Removes the need for manual _version.py file, instead it will be autogenerated.
- Refactored docs to fit the new structure, as well as adding new ones brought over from the skeleton
- Fixed all the warnings in docs builds
- Remove explicit support for Python2 (This involved no functional changes, but I removed it from the supported list in `pyproject.toml`)
- Use `pyproject.toml` for builds, where possible. Remove all unnecessary parts from `setup.py`
- Move Python code to `/src/cothread`
- Fix an issue discovered in `pvtree.py` between EPICS 3 and 7

Fixes #28

### CODE REVIEW

**Comment by Araneidae** [comment]
@coretl , I'm afraid I cannot constructively review this; can you please give it a sanity check?

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    "python.linting.pylintEnabled": false,
+    "python.linting.flake8Enabled": true,
+    "python.linting.mypyEnabled": true,
+    "python.linting.enabled": true,
+    "python.testing.pytestArgs": [
+        "--cov=python3_pip_skeleton",
```
Comment (coretl) [inline .vscode/settings.json:7]: Should be cothread

File: LICENSE
Comment (coretl) [inline LICENSE:1]: @Araneidae this switches to Apache like the rest of DLS modules, is that ok?

**Review by coretl — COMMENTED** [review]
Looks reasonable, just a couple of points

File: LICENSE
Comment (Araneidae) [reply LICENSE:1]: Sure

File: .vscode/settings.json (hunk 1 above)
Comment (AlexanderWells-diamond) [reply .vscode/settings.json]: Resolved in https://github.com/dls-controls/cothread/pull/41/commits/c49161003423171e6b7f700dc5acb24c8d7fd9f7

**Review by coretl — APPROVED** [review]

**Pull request merged**